### PR TITLE
perf: improve IntegerList API to avoid allocations

### PR DESCRIPTION
**Pull request by DaniPopes**
`RoaringBitmap` is constructed using iterators, don't force callers to construct `IntegerList` using slices.

### CODE REVIEW

File: crates/prune/prune/src/segments/user/history.rs [hunk 1]
```diff
@@ -151,7 +151,7 @@ where
         } else {
             cursor.upsert(
                 RawKey::new(key),
-                RawValue::new(BlockNumberList::new_pre_sorted(higher_blocks)),
+                RawValue::new(BlockNumberList::new_pre_sorted(higher_blocks())),
```
Comment (shekhirin) [inline crates/prune/prune/src/segments/user/history.rs:154]: why do this and call `higher_blocks` two times?

Comment (DaniPopes) [reply crates/prune/prune/src/segments/user/history.rs:154]: iterator vs collecting

Comment (shekhirin) [reply crates/prune/prune/src/segments/user/history.rs:154]: `blocks` can be up to 2000, is iterating over it two times faster than allocating once?

Comment (DaniPopes) [reply crates/prune/prune/src/segments/user/history.rs]: Probably not

**Review by shekhirin — APPROVED** [review]
LGTM, falling clippy

**Pull request merged**